File: Julia/5_EGM/rhs_euler.jl
"""
オイラー方程式の右辺を計算して返す関数

### インプット
`a::Float64`: 貯蓄水準
`params::Models`: パラメータを含む構造体

### アウトプット
残差
"""
function rhs_euler(a::Float64, params::Models)

    # 老年期の限界効用
    mu = mu_CRRA((1.0 + params.rent)*a, params.γ)

    # オイラー方程式の右辺
    return params.β*(1.0 + params.rent)*mu

end

File: Matlab/2_7_EGM/rhs_euler.m
function rhs = rhs_euler(asset)
% Function rhs_euler
%  [residual] = rhs_euler( a )
%
% 目的:
% オイラー方程式の右辺を計算.
%
% グローバル変数: beta, gamma, rent

global beta gamma rent

% 予算制約から消費を計算
cons = (1.0+rent)*asset;

% 老年期の限界効用
mu = mu_CRRA(cons, gamma);

% オイラー方程式の右辺
rhs = beta*(1.0+rent)*mu;

return;
